src/io/iohk/scalanet/peergroup/MessageChannel.scala

```diff
@@ -13,22 +13,24 @@ private[scalanet] class MessageChannel[A, MessageType](peerGroup: PeerGroup[A])(
 
   private val log = LoggerFactory.getLogger(getClass)
 
-  private val subscribers = new Subscribers[MessageType]()
+  private val subscribers = new Subscribers[(A, MessageType)]()
 
-  val inboundMessages: Observable[MessageType] = subscribers.messageStream
+  val inboundMessages: Observable[(A, MessageType)] = subscribers.messageStream
```

Comment:
As this value is exposed, use concrete type instead of the tuple.

Reply:
+1

Reply:
Don't have a name for it. Will give it some thought.